Neos.Flow/Classes/Http/Helper/ResponseInformationHelper.php

```diff
@@ -194,6 +206,21 @@ public static function makeStandardsCompliant(ResponseInterface $response, Reque
             $response = $response->withoutHeader('Content-Length');
         }
 
+        #
+        # Create a 304 response with an empty body and copy only the required headers
+        # from the original response as is required by RFC 7232 section 4.1
+        #
+        if ($response->getStatusCode() == 304) {
+            $keepHeaders = ['Cache-Control', 'Content-Location', 'Date', 'ETag', 'Expires', 'Vary'];
```

Comment:
Is this really supposed to be a safe-list and strip all other headers?

> The server generating a 304 response MUST generate any of the
   following header fields that would have been sent in a 200 (OK)
   response to the same request: Cache-Control, Content-Location, Date,
   ETag, Expires, and Vary.

I read that those headers MUST at least be added from the original 200 response. I don't read that others should be removed, though?
In case I'm wrong: At least, should we maybe also allow custom headers `X-*`?

Reply:
> Since the goal of a 304 response is to minimize information transfer
   when the recipient already has one or more cached representations, a
   sender SHOULD NOT generate representation metadata other than the
   above listed fields unless said metadata exists for the purpose of
   guiding cache updates (e.g., Last-Modified might be useful if the
   response does not have an ETag field).

I read that other header SHOULD be removed.

Reply:
Maybe we could add a configuration for that

Reply:
We could also ignore the header part and only empty the response body. That would avoid the need of having a configuration for that.

Reply:
@albe I decided to leave the headers in and only empty the response body for 304 responses. 
SHOULD allows this and this cause less problems and discussions then a whitelisting approach.